src/components/SharedComponents/OverviewCounts.tsx

```diff
@@ -1,32 +1,39 @@
-// @flow
-
 import {
   ActivityIndicator, Body2, Heading6, INatIcon,
 } from "components/SharedComponents";
 import { Pressable, View } from "components/styledComponents";
 import { t } from "i18next";
 import * as React from "react";
+import type { PressableProps } from "react-native";
 import colors from "styles/tailwindColors";
 
-type Props = {
-  counts: Object,
-  onObservationPressed: Function,
-  onSpeciesPressed: Function,
-  onMembersPressed: Function
+interface Counts {
+  observations_count: number;
+  species_count: number;
+  identifications_count: number;
+  members_count: number;
+  journal_posts_count: number;
+}
+
+interface Props {
+  counts: Counts;
+  onObservationPressed: PressableProps["onPress"];
+  onSpeciesPressed: PressableProps["onPress"];
+  onMembersPressed: PressableProps["onPress"];
 }
 
-type CountProps = {
-  count: number,
-  icon: string,
-  label: string
+interface CountProps {
+  count: number;
+  icon: string;
+  label: string;
 }
 
-type CountPressableProps = {
-  accessibilityLabel: string,
-  count: number,
-  icon: string,
-  label: string,
-  onPress?: Function
+interface CountPressableProps {
+  accessibilityLabel: string;
+  count: number;
+  icon: string;
+  label: string;
+  onPress?: PressableProps["onPress"];
```

Comment:
This is a new pattern to the codebase it seems to me.
We have several places where we propagate onPress down to a Pressable. We do use `( _event: GestureResponderEvent ) => void` in several places but also `() => void` if we are sure the caller does not need `event`.
Is this here preferable overall? Or is it easier to use this intermittently, in the sense of while we are typing from the tips of the tree, so that we don't have to check if the prop can actually be null or undefined? I would assume in the places where we typed it as a function we took care to check that neither null nor undefined is propped in.

Reply:
[Indexed access types](https://www.typescriptlang.org/docs/handbook/2/indexed-access-types.html) aren't a new pattern in the codebase:

<img width="412" height="435" alt="Screenshot 2025-12-24 at 6 12 14 AM" src="https://github.com/user-attachments/assets/1d9d1fbe-162c-4f1e-aec4-13db7d976254" />

I only did it because it's terser than writing out `null | ((event: GestureResponderEvent) => void) | undefined;` here and it's a basic feature of TypeScript so I didn't think there would be any controversy.